lib/rack/timeout/legacy.rb

```diff
@@ -10,23 +10,24 @@ module Rack::Timeout::ClassLevelProperties
   module ClassMethods
     attr_accessor :service_timeout, :wait_timeout, :wait_overtime, :service_past_wait
     alias_method :timeout=, :service_timeout=
+  end
 
+  module CallWithDeprecation
     [ :service_timeout=,
       :timeout=,
       :wait_timeout=,
       :wait_overtime=,
       :service_past_wait=,
     ].each do |isetter|
-      setter = instance_method(isetter)
-      define_method(isetter) do |x|
+      define_method(isetter) do |*|
         defined?(Rails) or warn "`Rack::Timeout.#{isetter}`: class-level settings are deprecated. See README for examples on using the middleware initializer instead."
-        setter.bind(self).call(x)
+        super
```

Comment:
I appreciate all your effort here. Maybe it's hubris but I believe I would have known to use `super`. So I'd advise heroku against merging this particular diff until I can review it.

To be fair, since it involves deprecation, maybe just follow up on the deprecation and remove the whole damn thing once and for all and issue a new version?